Chat/assignment-chat/client-chat/src/main/java/ru/assignment/net/Sender.java

```diff
@@ -33,34 +87,43 @@ public void startSender() {
                     writer.flush();
                     System.out.println("Sender write message");
                 } catch (IOException e) {
+                    System.out.println("exception");
                     e.printStackTrace();
                     break;
                 }
 
-                if (message.equalsIgnoreCase("disconnect" + "\n")) {
-                    System.out.println("exit sender disconnect");
+                if (message.equalsIgnoreCase("close" + "\n")) {
+                    System.out.println("exit sender close");
                     break;
                 }
             }
         } finally {
             if (this.isOpen()) {
-                closeSender();
+                close();
             }
         }
+*/
     }
 
-    public boolean isOpen() {
-        return isOpen;
+
+
+   public boolean isOpen() {
+      return isOpen;
     }
 
-    public void closeSender() {
-        System.out.println("closeSender");
+    public void disconnect() {
+        ChatClient.clientLogger.debug("Sender disconnect");
+        currentThread.interrupt();
         isOpen = false;
+    }
+
+    private void close() {
+        ChatClient.clientLogger.debug("Close sender");
         scanner.close();
         try {
             writer.close();
         } catch (IOException e) {
-            e.printStackTrace();
+            ChatClient.clientLogger.debug("Sender's writer close exception - {}",e);
```

Comment:
You shouldn't use {} when your are logging exception, please check signature:
log.debug(String message, Throwable t)

But here you are using log.debug(String message, Object param1);

Also, exceptions usually logged with WARN or ERROR or even higher levels